core/services/remote_access_services.py

```diff
@@ -47,17 +47,17 @@ def _get_vm_id():
         return vmconf.DEFAULT_VM_ID
 
     # Get VMID dynamically from metadata.
-    return metadata_services.get_metadata_param(
-        vmconf.METADATA_VM_ID_PARAM_NAME)
+    return str(metadata_services.get_metadata_param(
```

Comment:
why need to cast?

Reply:
`HMAC` does not support unicode strings.

Reply:
Add a comment then, here and below.

Reply:
Done.